core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -936,12 +936,10 @@ class KafkaApis(val requestChannel: RequestChannel,
       }
       erroneous.foreach { case (tp, data) => partitions.put(tp, data) }
 
-      var unconvertedFetchResponse: FetchResponse = null
-
-      def createResponse(throttleTimeMs: Int): FetchResponse = {
+      def createResponse(throttleTimeMs: Int, fetchResponse: FetchResponse): FetchResponse = {
```

Comment:
As there is a `convertedData`, it would be nice to keep using `fetchResponse`

Reply:
I will try to refactor it 🤔